src/core/reference/include/openvino/reference/fake_convert.hpp

```diff
@@ -0,0 +1,329 @@
+        out[i] = h.f;
+    }
+}
+
+template <typename T>
+void apply_scale_shift(T* out,
+                       const T* data,
+                       const T* scale,
+                       const T* shift,
+                       const Shape& data_shape,
+                       const Shape& scale_shape,
+                       const Shape& shift_shape,
+                       bool invert = false) {
+    OPENVINO_ASSERT(scale_shape == shift_shape, "Mismatch of `scale` and `shift` input shapes.");
+    const auto scale_size = shape_size(scale_shape);
+    const auto data_size = shape_size(data_shape);
+
+    if (scale_size == 1) {  // per tensor scale, probably for activation
+
+        T s = scale[0];
+        T o = shift[0];
+        if (invert) {
+            for (size_t j = 0; j < data_size; j++) {
+                out[j] = (data[j] + o) / s;
+            }
+        } else {
+            for (size_t j = 0; j < data_size; j++) {
+                out[j] = data[j] * s - o;  // o = quntized(o * s)
+            }
+        }
+        return;
+    }
+
+    if (scale_shape[0] == 1 && data_shape[1] == scale_shape[1]) {  // per channel scale for DW activations
+        size_t step = 1;
+        for (size_t i = 2; i < data_shape.size(); i++) {  // <batch_size, out_channels, in_channels, H, W> or
+            step *= data_shape[i];
+        }
+
+        for (size_t bs = 0; bs < data_shape[0]; bs++) {
+            for (size_t i = 0; i < scale_size; i++) {
+                T s = scale[i];
+                T o = shift[i];
+                if (invert) {
+                    for (size_t j = 0; j < step; j++) {
+                        out[j] = (data[j] + o) / s;
+                    }
+                } else {
+                    for (size_t j = 0; j < step; j++) {
+                        out[j] = data[j] * s - o;  // o = quntized(o * s)
+                    }
+                }
+                data += step;
+                out += step;
+            }
+        }
+        return;
+    }
+
+    OPENVINO_ASSERT(data_shape[0] == scale_size, "Shape mismatch in scale ");
+
+    // per channel scale for weights
+    size_t step = 1;
+    for (size_t i = 1; i < data_shape.size(); i++) {
+        step *= data_shape[i];
+    }
+
+    for (size_t i = 0; i < scale_size; i++) {
+        T s = static_cast<T>(scale[i]);
+        if (invert) {
+            for (size_t j = 0; j < step; j++) {
+                out[j] /= s;
+            }
+        } else {
+            for (size_t j = 0; j < step; j++) {
+                out[j] *= s;
+            }
+        }
+        data += step;
+        out += step;
+    }
+}
+}  // namespace fake_convert_details
+
+template <typename T>
+bool apply_conversion(const T* data, T* out, size_t element_count, const std::string& destination_type) {
+    auto inPtr = reinterpret_cast<const unsigned short*>(data);
+    auto outPtr = reinterpret_cast<unsigned short*>(out);
+    if (destination_type == "f8e5m2") {
```

Comment:
make some reference helper in hpp and cpp which will convert string to conversion mode (enum) and remove assertion here.
Don't put non-template code into template make template as think as possible.

Reply:
Converted to non a template function. No additional structure for the destination_mode for now, considered as a follow up refactor or simple replacement by ov element type when possible.